tests/privacy_simple_network.py

```diff
@@ -98,127 +99,35 @@
-
-    def moveToNonParticipants():
-        movedNode = participants[-1]
-        # popping off back of participants and need to push on the front of nonParticipants
-        nonParticipants.insert(0, movedNode)
-        del participants[-1]
-        return movedNode
-
-    def addToSg():
-        node = moveToParticipants()
-        Utils.Print("Take a non-participant and make a participant. Now there are {} participants and {} non-participants".format(len(participants), len(nonParticipants)))
-        toAddNum = cluster.getParticipantNum(node)
-        return security_group([toAddNum])
-
-    def remFromSg():
-        node = moveToNonParticipants()
-        Utils.Print("Take a participant and make a non-participant. Now there are {} participants and {} non-participants".format(len(participants), len(nonParticipants)))
-        toRemoveNum = cluster.getParticipantNum(node)
-        return security_group(removeNodeNums=[toRemoveNum])
+    securityGroup = cluster.getSecurityGroup()
 
     Utils.Print("Add all producers to security group")
-    publishTrans = security_group([x for x in range(pnodes)])
-    verifySecurityGroup(publishTrans)
+    securityGroup.editSecurityGroup([cluster.getNodes()[x] for x in range(pnodes)])
+    securityGroup.verifySecurityGroup()
 
     cluster.reportInfo()
 
-
+    Utils.Print("One by one, add each API Node to the security group")
     # one by one add each nonParticipant to the security group
-    while len(nonParticipants) > 0:
-        publishTrans = addToSg()
-        verifySecurityGroup(publishTrans)
+    while len(securityGroup.nonParticipants) > 0:
+        securityGroup.addToSecurityGroup()
+        securityGroup.verifySecurityGroup()
         cluster.reportInfo()
 
 
+    Utils.Print("One by one, remove each API Node from the security group")
     # one by one remove each (original) nonParticipant from the security group
-    while len(participants) > pnodes:
-        publishTrans = remFromSg()
-        verifySecurityGroup(publishTrans)
+    while len(securityGroup.participants) > pnodes:
+        securityGroup.remFromSecurityGroup()
+        securityGroup.verifySecurityGroup()
         cluster.reportInfo()
 
 
     # if we have more than 1 api node, we will add and remove all those nodes in bulk, if not it is just a repeat of the above test
     if len(apiNodes) > 1:
-        # add all the api nodes to security group at once
-        toAdd = []
-        for apiNode in nonParticipants:
-            participantNum = cluster.getParticipantNum(apiNode)
-            toAdd.append(participantNum)
-        participants.extend(nonParticipants)
-        nonParticipants = []
-
-        Utils.Print("Add all api nodes to security group")
-        publishTrans = security_group(addNodeNums=toAdd)
-        verifySecurityGroup(publishTrans)
+        Utils.Print("Add all api nodes to security group at the same time")
+        securityGroup.editSecurityGroup(addNodes=securityGroup.nonParticipants)
```

Comment:
now I see why you used `copy` inside `editSecurityGroup`. First of all I think proper behavior is to use shallow copy `copy.copy` and not deep copy. Second, maybe we can add helper method like `moveAllToSecurityGroup` to avoid using internal structures as parameters?

Reply:
changed all deepcopy to copy. Added a local function to only copy if it is one of our containers